EventsExpress/ClientApp/src/actions/event/event-add-action.js

```diff
@@ -26,15 +26,17 @@ export default function add_event() {
     }
 }
 
-export function edit_event(data) {
+export function edit_event(data, onError, onSuccess) {
     return async dispatch => {
         dispatch(getRequestInc());
         let response = await api_serv.editEvent(data);
-        if (!response.ok) {
-            throw new SubmissionError(await buildValidationState(response));
+        dispatch(getRequestDec());
+        if (!response.ok && onError && typeof onError === 'function') {
```

Comment:
This code can process success messages even if the response is not successful. Please split the check of the response and whether the parameter was specified. Also, you don't need to check for the function type - it's internal code and it will always be a function.

Reply:
Agree